tests/components/zha/test_discover.py

```diff
@@ -45,11 +61,25 @@
             ent for ent in entity_ids if ent.split(".")[0] in zha_const.COMPONENTS
         }
 
-        event_channels = {
-            arg[0].cluster_id
-            for arg, kwarg in cr_ch.call_args_list
-            if kwarg.get("channel_class") == EventRelayChannel
+        zha_dev = zha_gateway.get_device(zigpy_device.ieee)
+        event_channels = {  # pylint: disable=protected-access
+            ch.id for ch in zha_dev._relay_channels.values()
         }
 
         assert zha_entities == set(device["entities"])
         assert event_channels == set(device["event_channels"])
+
+        entity_map = device["entity_map"]
+        for calls in _dispatch.call_args_list:
+            discovery_info = calls[0][2]
+            unique_id = discovery_info["unique_id"]
+            channels = discovery_info["channels"]
+            component = discovery_info["component"]
+            key = (component, unique_id)
+            entity_id = entity_registry.async_get_entity_id(component, "zha", unique_id)
+
+            assert key in entity_map
+            assert entity_id is not None
+            no_tail_id = NO_TAIL_ID.sub("", entity_map[key]["entity_id"])
+            assert entity_id.startswith(no_tail_id)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4896509)